php/PHPID.php

```diff
@@ -179,6 +179,12 @@ private function updateInterfaceIndex($interface, $implementation)
 
     private function saveChild($index_file, $child)
     {
+        $index_directory = dirname($index_file);
+
+        if (is_dir($index_directory)) {
```

Comment:
Oops, I just realized there is a bug here... this should read `!is_dir(...)`. Right now the condition will always fail with a missing index.